framework/src/fwk_module.c

```diff
@@ -449,6 +450,86 @@ int fwk_module_start(void)
     return FWK_SUCCESS;
 }
 
+static int fwk_module_stop_elements(struct fwk_module_context *module_ctx)
+{
+    int status;
+    const struct fwk_module *module;
+    unsigned int element_idx;
+
+    module = module_ctx->desc;
+
+    for (element_idx = 0; element_idx < module_ctx->element_count;
+         element_idx++) {
+
+        if (module->stop != NULL) {
+            status = module->stop(
+                        fwk_id_build_element_id(module_ctx->id, element_idx));
+            if (!fwk_expect(status == FWK_SUCCESS)) {
+                FWK_LOG_CRIT(fwk_module_err_msg_func, status, __func__);
+                return status;
+            }
+        }
+
+        module_ctx->element_ctx_table[element_idx].state =
+            FWK_MODULE_STATE_SUSPENDED;
+    }
+
+    return FWK_SUCCESS;
+}
+
+static int fwk_module_stop_module(struct fwk_module_context *module_ctx)
+{
+    int status;
+    const struct fwk_module *module;
+
+    module = module_ctx->desc;
+
+    if (module->stop != NULL) {
+        status = module->stop(module_ctx->id);
+        if (!fwk_expect(status == FWK_SUCCESS)) {
+            FWK_LOG_CRIT(fwk_module_err_msg_func, status, __func__);
+            return status;
+        }
+    }
+
+    module_ctx->state = FWK_MODULE_STATE_SUSPENDED;
+
+    return fwk_module_stop_elements(module_ctx);
+}
+
+static int stop_modules(void)
+{
+    int status;
+    unsigned int module_idx;
+    struct fwk_module_context *module_ctx;
+
+    for (module_idx = 0; module_idx < FWK_MODULE_IDX_COUNT; module_idx++) {
+        module_ctx = &fwk_module_ctx.module_ctx_table[module_idx];
+        status = fwk_module_stop_module(module_ctx);
+        if (status != FWK_SUCCESS)
+            return status;
+    }
+
+    return FWK_SUCCESS;
+}
+
+int fwk_module_stop(void)
+{
+    int status;
+
+    if (!fwk_module_ctx.initialized) {
+        FWK_LOG_CRIT(fwk_module_err_msg_func, FWK_E_STATE, __func__);
+        return FWK_E_STATE;
+    }
+
+    fwk_module_ctx.stage = MODULE_STAGE_STOP;
+    status = stop_modules();
+    if (status != FWK_SUCCESS)
```

Comment:
We need the brackets around to body of the if statement to avoid MISRA violation.
"The body of the "then" branch of the "if" statement is not a compound statement."